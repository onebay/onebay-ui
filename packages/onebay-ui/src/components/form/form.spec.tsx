import { mount } from '@vue/test-utils'
import Form from './index'

const factory = (values = {}, slots = { default: [] }) => {
  return mount(Form, {
    slots,
    props: { ...values },
  })
}

describe('Form Snap', () => {
  it('render initial Form', () => {
    const wrapper = factory()
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render initial Form', () => {
    const wrapper = factory({ reportSubmit: true })
    expect(wrapper.element).toMatchSnapshot()
  })
})

describe('Form Event', () => {
  it('Form onSubmit', () => {
    const onSubmit = jest.fn()
    const wrapper = factory({ onSubmit: onSubmit })
    wrapper.find('.ob-form').trigger('submit')
    expect(onSubmit).toBeCalled()
  })

  it('Form onReset', () => {
    const onReset = jest.fn()
    const wrapper = factory({ onReset: onReset })
    wrapper.find('.ob-form').trigger('reset')
    expect(onReset).toBeCalled()
  })
})
