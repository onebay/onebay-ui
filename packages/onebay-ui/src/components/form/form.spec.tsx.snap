// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form Snap render initial Form 1`] = `
<form
  class="ob-form"
>
  
  
</form>
`;

exports[`Form Snap render initial Form 2`] = `
<form
  class="ob-form"
>
  
  
</form>
`;
